src/qbe_ir_generator.cpp

```diff
@@ -124,14 +124,28 @@ void QbeIrGenerator::Visit(const IfStmtNode& if_stmt) {
   if_stmt.predicate->Accept(*this);
   int predicate_num = num_recorder.NumOfPrevExpr();
   int label_num = NextLabelNum();
-  std::string start_label = PrefixLabel("if", label_num);
+  std::string if_label = PrefixLabel("if", label_num);
```

Comment:
```suggestion
  std::string then_label = PrefixLabel("then", label_num);
```
Using `then_label` would be more expressive, indicating that this label marks the start of the "then" part.

Reply:
@Lai-YT 
I think using `then` as the first statement body member for `IfStmtNode` is a great idea because it has additional context.

```
if_stmt.then->Accept(*this);
```

However, I think we should keep `if_label` for this case from the perspective of QBE IR. It would be more straightforward looking at the following IR code. I think `@if.1` pairs nicely with `@else.1`, but `@then.1` doesn't provide enough context in the assembly code since there's only jump instruction at the IR level. What do you think?

Suggested change:
```
jnz %.5, @then.1, @else.1
@then.1
  %.6 =w copy 0
  storew %.6, %.1
  jmp @end.1
@else.1
  %.7 =w loadw %.1
  %.8 =w copy 1
  %.9 =w add %.7, %.8
  storew %.9, %.1
@end.1
```

Current:
```
jnz %.5, @if.1, @else.1
@if.1
  %.6 =w copy 0
  storew %.6, %.1
  jmp @end.1
@else.1
  %.7 =w loadw %.1
  %.8 =w copy 1
  %.9 =w add %.7, %.8
  storew %.9, %.1
@end.1
```

Reply:
Having this may be confusing in the code, right? 😂 
```
std::string then_label = PrefixLabel("if", label_num);
```

Reply:
In my humble opinion, in an _if-then-else_ statement, _if_ refers to the condition, while _then_ and _else_ refer to the block statements.
That's why I propose labels like `@then.1` and `@else.1` to maintain clarity.
However, I understand your concern. How about choosing one of the following options:
- Add a comment `# if` above the evaluation of the predicate.
- Change the label of the "then" part to `@if_then.1`.

Reply:
I like option 1 better because we can increase readability in both source code and IR.
So, I will use `then_label` in the source code and add a comment above the predicate in the IR. 👍🏼